tests/serializers/test_complex.py

```diff
@@ -0,0 +1,39 @@
+import math
+
+import pytest
+
+from pydantic_core import SchemaSerializer, core_schema
+
+
+@pytest.mark.parametrize(
+    'value,expected',
+    [
+        (complex(-1.23e-4, 567.89), '-0.000123+567.89j'),
+        (complex(0, -1.23), '-1.23j'),
+        (complex(1.5, 0), '1.5+0j'),
+        (complex(1, 2), '1+2j'),
+        (complex(0, 1), '1j'),
+        (complex(0, 1e-500), '0j'),
+        (complex(-float('inf'), 2), '-inf+2j'),
+        (complex(float('inf'), 2), 'inf+2j'),
+        (complex(float('nan'), 2), 'NaN+2j'),
```

Comment:
It would be interesting to discuss how these interact with the `ser_json_inf_nan` setting (if at all). Maybe they shouldn't, given these are always strings.

Reply:
I thought about another option `allow_inf_nan` and eventually decided not to deal with it at all because I'm not really sure how this should apply to complex numbers and implementing anything with any reasoning will probably face objections from someone else.

Even though `re` and `im` parts are both `float`, the entire value is still a complex object. Rejecting something like `complex(float('inf'), 2)` kind of makes sense if we think of complex objects as "containers of float numbers", but at the same time it doesn't sound that natural because a complex number is... a complex number, a complete entity, not just "something with two float attributes, `re` and `im`".

Reply:
I broadly agree with this, but what happens when the imaginary component is zero? That's the only case which I think could be confusing if it differs from a regular float.

Reply:
Ah I see, we get a `+0j` part still. So yes let's not worry about that setting.